components/event-publisher-proxy/pkg/subscribed/helpers.go

```diff
@@ -72,10 +91,49 @@ func AddUniqueEventsToResult(eventsSubSet []Event, uniqEvents map[Event]bool) ma
 }
 
 // FilterEventTypeVersions returns a slice of Events:
+// 1. if the eventType matches the format for standard type matching: <event-name>.<version>
+// E.g. order.created.v0
+// 2. if the eventType matches the format for exact type matching: <eventTypePrefix><appName>.<event-name>.<version>
+// E.g. sap.external.custom.exampleapp.order.created.v0
+func FilterEventTypeVersions(appName string, subscription *eventingv1alpha2.Subscription) []Event {
+	events := make([]Event, 0)
+	for _, eventType := range subscription.Spec.Types {
+		if subscription.Spec.TypeMatching == eventingv1alpha2.TypeMatchingExact {
+			// in case of type matching exact, we have app name as a part of event type
+			appNameWithDots := "." + appName + "."
+			if strings.Contains(eventType, appNameWithDots) {
+				eventTypeVersionArr := strings.Split(eventType, appNameWithDots)
+				eventTypeVersion := eventTypeVersionArr[len(eventTypeVersionArr)-1]
+				event := buildEvent(eventTypeVersion)
+				events = append(events, event)
+			}
+		} else {
+			// in case of type matching standard, the source must be app name
+			if appName == subscription.Spec.Source {
+				event := buildEvent(eventType)
+				events = append(events, event)
+			}
+		}
+	}
+	return events
+}
+
+// it receives event and type version, e.g. order.created.v1 and returns {Name: order.created, Version: v1}
+func buildEvent(eventTypeAndVersion string) Event {
```

Comment:
(nit) can you write a unit test for this function as well. I'm aware that its called in the previous function and its unit tested, but would be nice to unit test the smallest block as well

Reply:
done